File: solutions/go/bird-watcher/1/bird_watcher.go
package birdwatcher

// TotalBirdCount return the total bird count by summing
// the individual day's counts.
func TotalBirdCount(birdsPerDay []int) int {
	var sum int
	for i := 0; i < len(birdsPerDay); i++ {
		sum = sum + birdsPerDay[i] 
	}
	return sum
}

// BirdsInWeek returns the total bird count by summing
// only the items belonging to the given week.
func BirdsInWeek(birdsPerDay []int, week int) int {
	starDate := (week - 1) * 7
	endDate := week * 7

	var weekSum int
	for i := starDate; i < endDate; i++ {
		weekSum += birdsPerDay[i]
	}
	return weekSum
}

// FixBirdCountLog returns the bird counts after correcting
// the bird counts for alternate days.
func FixBirdCountLog(birdsPerDay []int) []int {
	for i := 0; i < len(birdsPerDay); i ++ {
		if (i + 1) % 2 != 0 {
			birdsPerDay[i] += 1
		}
	}

	return birdsPerDay
}

File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
        return birdsPerDay;
    }

    public int getToday() {
        int lengthOfArray = birdsPerDay.length;
        return birdsPerDay[lengthOfArray-1];
    }

    public void incrementTodaysCount() {
        int lengthOfArray = birdsPerDay.length;
        birdsPerDay[lengthOfArray-1] += 1;
    }

    public boolean hasDayWithoutBirds() {
        int count = 0;
        for (int i = 0; i < birdsPerDay.length; i ++){
            if (birdsPerDay[i] == 0){
                count ++;
            }
        }
        return count != 0;
    }

    public int getCountForFirstDays(int numberOfDays) {
        int sum = 0;
        for (int i = 0; i < numberOfDays && i < birdsPerDay.length; i++){
            sum += birdsPerDay[i];
        }
        return sum;
    }

    public int getBusyDays() {
        int count = 0;
        for (int i = 0; i < birdsPerDay.length; i ++){
            if (birdsPerDay[i] >= 5){
                count ++;
            }
        }

        return count;
    }
}
